### PR TITLE
TDL-14339 added configurable lookback window

### PR DESCRIPTION
**Pull request by namrata270998**
# Description of change
- Added configurable lookback window and a default value in case not passed in the config.
- Added unittests for the same
- Increased the default lookback window size to 600(10 minutes) double than the current.

**NOTE**: The build is failing as we need to add a config parameter on the cci side.

# Manual QA steps
 - Check that the start_date starts from the configurable lookback window
 - Check that when lookback not configured, the tap takes a default value
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

**Review by savan-chovatiya — APPROVED** [review]

File: tap_stripe/__init__.py [hunk 1]
```diff
@@ -107,7 +107,7 @@
 
 # NB: These streams will only sync through once for creates, never updates.
 IMMUTABLE_STREAMS = {'balance_transactions', 'events'}
-IMMUTABLE_STREAM_LOOKBACK = 300 # 5 min in epoch time, Stripe accuracy is to the second
+IMMUTABLE_STREAM_LOOKBACK = 600 # 10 min in epoch time, Stripe accuracy is to the second
```
Comment (dbshah1212) [inline tap_stripe/__init__.py:110]: Add Jira ticket in the comment for this change.

Comment (namrata270998) [reply tap_stripe/__init__.py]: Added the jira ticket in the comment

File: tap_stripe/__init__.py [hunk 2]
```diff
@@ -107,7 +107,7 @@
 
 # NB: These streams will only sync through once for creates, never updates.
 IMMUTABLE_STREAMS = {'balance_transactions', 'events'}
-IMMUTABLE_STREAM_LOOKBACK = 300 # 5 min in epoch time, Stripe accuracy is to the second
+IMMUTABLE_STREAM_LOOKBACK = 600 # 10 min in epoch time, Stripe accuracy is to the second. Jira link: https://jira.talendforge.org/browse/TDL-14339
```
Comment (dbshah1212) [inline tap_stripe/__init__.py:110]:
```suggestion
IMMUTABLE_STREAM_LOOKBACK = 600 # 10 min in epoch time, Stripe accuracy is to the second. Issue: TDL-14339
```

Comment (namrata270998) [reply tap_stripe/__init__.py]: Updated

**Review by kspeer825 — COMMENTED** [review]
The ticket description mentions `balance_transactions` stream only, but this lookback window appears to be used for the `events` stream as well. Is it expected that that window should be changed for `events`?

**Review by kspeer825 — CHANGES_REQUESTED** [review]
Could you add a new parameterized tap-tester test that runs once against all streams using the default lookback, and then again with a value provided in the config. The goal of this test is to prove the configurable property can be handled by the tap when it is ran with the orchestrator and target. 
Here is an example to help with implementation. 
https://github.com/singer-io/tap-google-ads/blob/qa/configurable-props/tests/test_google_ads_conversion_window.py
Please work with @RushT007 on this implementation if he is available.

File: tests/unittests/test_lookback_window.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+@mock.patch("tap_stripe.sync_sub_stream")
+@mock.patch("tap_stripe.get_bookmark_for_stream", return_value = start_date_before_lookback)
+class TestLookbackWindow(unittest.TestCase):
+
+    def test_default_value_lookback(self, mock_get_bookmark_for_stream, mock_sync_substream, mock_dt_to_epoch, mock_epoch_to_dt, mock_get, mock_metadata_map, mock_get_catalog_entry, mock_paginate, mock_convert_dict_to_stripe_object, mock_reduce_foreign_keys):
+        '''Verify that the lookback window is 600 by default if nothing is provided in the config.'''
+        config = { "client_secret": "test_secret", "account_id": "test_account", "start_date": "test_start_date"}
+        Context.config = config
+        Context.new_counts['balance_transactions'] = 1
+        sync_stream("balance_transactions")
```
[the remark this answers is not in the document]
Comment (namrata270998) [reply tests/unittests/test_lookback_window.py:35]: We are not intending for test the `events` stream, hence updated the code

**Review by kspeer825 — APPROVED** [review]

**Comment by namrata270998** [comment]
Closing this PR as the code from here is included in the [PR](https://github.com/singer-io/tap-stripe/pull/138)

**Pull request closed without merging**